Adding Support for AWS EventBridge for Shopify Webhooks

usmanpakistan opened the pull request.

usmanpakistan commented:
@Kyon147 This PR resolve issue reported previously on the legacy laravel-shopify package. 
It enables a user to provide an EventBridge ARN while subscribing to Webhooks. You can refer to your comment here for more detail
https://github.com/gnikyt/laravel-shopify/issues/1113#issuecomment-1086071686

usmanpakistan commented:
@Kyon147 Any update on this PR

Jamesking56 commented:
> @Kyon147 Any update on this PR

Your PR is still failing tests, resolve those first please

Kyon147 commented:
Hey @usmanpakistan 

As @Jamesking56 mentions please fix the failing tests and I can re-review once they are good.

usmanpakistan commented:
@Jamesking56 @Kyon147 can I get some help with fixing these tests. I am not well versed with writing them. I have tried a lot to fix the tests but these are still failing

naqashm commented:
Hi,

This looks like a great addition. Do you have an estimated timeline for when this will go live or be merged? We are currently using AWS EventBridge as well, so this support would be very helpful for us.

Thank you.

Kyon147 commented:
I'll take a look at the weekend for this as I need to look into the new offline tokens

Jamesking56 commented:
There's a lot of duplication in this PR, can you take a look at reducing the duplication @usmanpakistan ?

usmanpakistan commented:
@Jamesking56 I have updated the PR. The duplicated code was replaced and if else statements were replaced with dynamically build queries. I have also updated the tests and GraphQL fixtures. The PR has successfully passed all the automated checks. @Kyon147

naqashm commented:
Hi @Kyon147, I wanted to follow up on this PR. Could you please share any updates when convenient?
